flang/lib/Optimizer/Builder/Runtime/Command.cpp

```diff
@@ -19,3 +20,46 @@ mlir::Value fir::runtime::genCommandArgumentCount(fir::FirOpBuilder &builder,
       fir::runtime::getRuntimeFunc<mkRTKey(ArgumentCount)>(loc, builder);
   return builder.create<fir::CallOp>(loc, argumentCountFunc).getResult(0);
 }
+
+void fir::runtime::genGetCommandArgument(fir::FirOpBuilder &builder,
+                                         mlir::Location loc, mlir::Value number,
+                                         mlir::Value value, mlir::Value length,
+                                         mlir::Value status,
+                                         mlir::Value errmsg) {
+  auto argumentValueFunc =
+      fir::runtime::getRuntimeFunc<mkRTKey(ArgumentValue)>(loc, builder);
+  auto argumentLengthFunc =
+      fir::runtime::getRuntimeFunc<mkRTKey(ArgumentLength)>(loc, builder);
+
+  auto isPresent = [&](mlir::Value val) -> bool {
+    auto definingOp = val.getDefiningOp();
+    if (auto cst = mlir::dyn_cast<fir::AbsentOp>(definingOp))
+      return false;
+    return true;
+  };
+
+  if (isPresent(value) || status || isPresent(errmsg)) {
+    llvm::SmallVector<mlir::Value> args = fir::runtime::createArguments(
+        builder, loc, argumentValueFunc.getType(), number, value, errmsg);
```

Comment:
One thing to possibly consider in the future is to generate optional code that tests `number` to make sure it is inbounds here and issue a snazzy error message. Eventually, flang will probably want to generate runtime code to check that offsets are inbounds for arrays, etc.